protostar/commands/test/testing_live_logger.py

```diff
@@ -81,9 +82,10 @@ def log(
                         if (
                             self.exit_first
                             and shared_tests_state.any_failed_or_broken()
+                            and not isinstance(test_result, NONNEGATIVE_RESULTS)
```

Comment:
Is this necessary? Isn't this already taken care of in [protostar/testing/test_shared_tests_state.py](https://github.com/software-mansion/protostar/pull/1326/files#diff-ced3a88b3f75b799756c02bea11369baf57085419b57fe9ceade5bd6d6060229)'s `put_result`?

Reply:
I believe this is a reason of the issue? Because variable may be set to true but the result may be on the queue? is it @Radinyn ?

My question here is, why we need the any_failed_or_broken variable in the first place? Wouldn't be better to keep the control in the parent process here and just terminate other processes when first fail is reported? Every piece of shared state removed is a benefit for the codebase 🏗️

Reply:
Last time we were working at this code we couldn't find a way to do this without delegating some more logic to the logger, I think that removing this shared flag isn't as important as preventing this.